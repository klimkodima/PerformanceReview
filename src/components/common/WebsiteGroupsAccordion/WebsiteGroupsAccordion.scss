.custom-accordion {
  all: unset;

  &__title {
    font-size: 24px;
    padding-left: 30px;
    padding-top: 2px;
    color: #777777;
    padding-bottom: 30px;
  }

  &__form {
    display: flex;
    margin: 12px 0 0 0;

    .custom-accordion-summary {
      padding: 0;
      min-height: 16px;
      display: flex;
      justify-content: flex-start;
      align-items: flex-start;

      .MuiAccordionSummary-content {
        margin: 0;
      }

      .MuiAccordionSummary-expandIconWrapper.Mui-expanded {
        transform: rotate(90deg);
      }

      .MuiFormControlLabel-root {
        margin: 0;
      }

      .arrow-icon {
        color: #939393;
        width: 24px;
        height: 24px;
      }

      .title-icon {
        width: 40px;
        height: 40px;
        color: #939393;
      }
    }
  }

  &__details {
    padding: 0;

    .MuiBox-root {
      margin-left: 50px;
    }

    &.MuiAccordionDetails-root {
      padding: 10px 0 0 0;
    }
  }
}
