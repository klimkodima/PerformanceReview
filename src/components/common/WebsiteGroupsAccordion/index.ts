export { default as WebsiteGroupsAccordion } from './WebsiteGroupsAccordion';
