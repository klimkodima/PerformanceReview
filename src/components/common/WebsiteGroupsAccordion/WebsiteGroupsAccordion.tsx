import { FC, ReactElement, memo, ReactNode } from 'react';

import { Accordion, AccordionSummary, AccordionDetails } from '@mui/material';

import './WebsiteGroupsAccordion.scss';

type CustomAccordionPropsType = {
  expanded: boolean;
  summary: ReactNode;
  expandIcon: ReactNode;
};

const WebsiteGroupsAccordion: FC<CustomAccordionPropsType> = memo(
  ({ expanded, summary, children, expandIcon }): ReactElement => (
    <Accordion
      disableGutters
      expanded={expanded}
      className='custom-accordion'
      data-testid='custom-accordion'
    >
      <div className='custom-accordion__form'>
        <AccordionSummary
          className='custom-accordion-summary'
          expandIcon={expandIcon}
        >
          {summary}
        </AccordionSummary>
      </div>
      <AccordionDetails className='custom-accordion__details'>
        {children}
      </AccordionDetails>
    </Accordion>
  )
);

export default WebsiteGroupsAccordion;

WebsiteGroupsAccordion.displayName = 'WebsiteGroupsAccordion';
