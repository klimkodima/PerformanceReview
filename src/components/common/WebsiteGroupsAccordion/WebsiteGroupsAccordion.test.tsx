import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import WebsiteGroupsAccordion from './WebsiteGroupsAccordion';

describe('WebsiteGroupsAccordion', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <WebsiteGroupsAccordion
          expanded={true}
          summary={<div>Hello</div>}
          expandIcon={<p>#</p>}
        >
          <div>inner component</div>
        </WebsiteGroupsAccordion>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('inner component must be expanded', () => {
    render(
      <WebsiteGroupsAccordion
        expanded={true}
        summary={<div>Hello</div>}
        expandIcon={<p>#</p>}
      >
        <div>inner component</div>
      </WebsiteGroupsAccordion>
    );
    expect(screen.getByTestId('custom-accordion')).toHaveClass('Mui-expanded');
  });
  test('inner component must be hidden', () => {
    render(
      <WebsiteGroupsAccordion
        expanded={false}
        summary={<div>Hello</div>}
        expandIcon={<p>#</p>}
      >
        <div>inner component</div>
      </WebsiteGroupsAccordion>
    );
    expect(screen.getByTestId('custom-accordion')).not.toHaveClass(
      'Mui-expanded'
    );
  });
});
