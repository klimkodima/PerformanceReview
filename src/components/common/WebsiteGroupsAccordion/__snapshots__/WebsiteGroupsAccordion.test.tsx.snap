// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebsiteGroupsAccordion renders correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded Mui-expanded custom-accordion css-1wc0u5o-MuiPaper-root-MuiAccordion-root"
  data-testid="custom-accordion"
>
  <div
    className="custom-accordion__form"
  >
    <div
      aria-expanded={true}
      className="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded custom-accordion-summary css-1pvvkxv-MuiButtonBase-root-MuiAccordionSummary-root"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiAccordionSummary-content Mui-expanded css-1betqn-MuiAccordionSummary-content"
      >
        <div>
          Hello
        </div>
      </div>
      <div
        className="MuiAccordionSummary-expandIconWrapper Mui-expanded css-yw020d-MuiAccordionSummary-expandIconWrapper"
      >
        <p>
          #
        </p>
      </div>
    </div>
  </div>
  <div
    className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
      >
        <div
          className="MuiAccordion-region"
          role="region"
        >
          <div
            className="MuiAccordionDetails-root custom-accordion__details css-15v22id-MuiAccordionDetails-root"
          >
            <div>
              inner component
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
